modules/standard/Regex.chpl

```diff
@@ -586,10 +586,13 @@ proc bytes.this(m:regexMatch) {
 }
 
 
+/* We hold a copy of pattern string/bytes in its serialized form inside
+ * this record.
+ */
 pragma "no doc"
 record chpl_serializeHelper {
   type exprType;
-  var pattern:exprType;
+  var pattern:(new exprType()).chpl__serialize().type;
```

Comment:
I'm not sure the current state of this, but at least in the past we didn't DCE expressions that were only used to compute a type. Can you verify this isn't resulting in an allocation?

Reply:
Hmm I'm pretty sure it doesn't but not 100% sure I'm looking in all the right places. The type is definitely resolved at compile time 
```C
typedef struct chpl_chpl_serializeHelper_string_s {
  chpl____serializeHelper pattern;
  qio_regex_options_t options;
} chpl_serializeHelper_string;
```
and I don't see any calls to `chpl__serialize*` in eg module init code or anywhere unexpected

Reply:
We have trouble with DCE of expressions computing a type when the type has a runtime component (arrays, domains)